components/src/dynamo/trtllm/llm_engine.py

```
@@ -354,11 +404,21 @@ def _metrics_poll_loop(self) -> None:
+                if kv_used is not None:
+                    self._latest_metrics[rank] = Metrics(kv_used_blocks=int(kv_used))
+                # Drive upstream Prometheus collector with the same snapshot.
+                # Mirrors the legacy publisher.py path; under attention-DP,
+                # later ranks overwrite earlier within a tick — same behavior
+                # as upstream trtllm-serve's openai_server.py:1067.
+                if self._metrics_collector is not None and hasattr(
+                    self._metrics_collector, "log_iteration_stats"
+                ):
+                    try:
+                        self._metrics_collector.log_iteration_stats(snap)
+                    except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)